### DISCUSSION TITLE
stitching drone video snapshots

### DISCUSSION THREAD

**Discussion by TByte007** [OpenStitching/stitching#80] [Q&A]
Is it possible to stitch images vertically like those ?
![00003](https://user-images.githubusercontent.com/13002165/194552238-d76b605a-e16e-4385-8031-09e80f6d7902.jpg)
![00002](https://user-images.githubusercontent.com/13002165/194552155-acaa3799-7795-4f24-9a11-db7a5e2e45d8.jpg)
![00001](https://user-images.githubusercontent.com/13002165/194552157-579e999a-4b15-4725-978a-3ff8ccdb1a85.jpg)

**Comment by lukasalexanderweber** [comment]
You should be. Try with 

`stitch * --wave_correct_kind vert`

or

`stitch * --wave_correct_kind no`

**Comment by TByte007** [comment]
> You should be. Try with
> 
> `stitch * --wave_correct_kind vert`
> 
> or
> 
> `stitch * --wave_correct_kind no`

Both produced this (vert just turned it horizontally) (sorry for the wrong title :) ). The third image is missing
![result](https://user-images.githubusercontent.com/13002165/194564997-a18c22ff-709c-481c-aff4-e191a5ca1b5e.jpg)

**Comment by lukasalexanderweber** [comment]
I have got a result using

`stitch * --confidence_threshold 0 --wave_correct_kind vert --detector sift`

Apperently the matching confidences between the first and the other images are not that good. And with the default detector orb the stitching would fail. sift seems to be better for your use case.

Feel free to follow the tutorial for more insights!

![result](https://user-images.githubusercontent.com/32765578/194578973-d5e59fcc-ee6c-4102-92e3-8aea6af472ad.jpg)

**Comment by TByte007** [comment]
> --confidence_threshold 0 --wave_correct_kind vert --detector sift

This works way better ! But should all the frames have to be matched to the first one ? I'm trying to create something like a map from the drone footage and most of the frames will have nothing to do with the first one.
Should I stitch the next frame to the result of the previous for example ?
Thank you for your time !

**Comment by lukasalexanderweber** [comment]
I'd rather be interested in your dataset. Is it a grid and you know the relation of the Images? E.g. geographic bounding boxes? How many Images are there? Do you know for sure that the Images overlap?

**Comment by TByte007** [comment]
> I'd rather be interested in your dataset. Is it a grid and you know the relation of the Images? E.g. geographic bounding boxes? How many Images are there? Do you know for sure that the Images overlap?

I'm testing it with a random 4k (downsized to 1024x768) drone video by taking snapshots and assuring overlap with the previous frame. In reality (when I have a drone) there would be GPS data too and by using it you can approximate the position of one image to another. But that would require a slightly different algorithm for stitch I guess (if you include the GPS data) ?
PS: I tested it with 12 images and it still works very well.
![result](https://user-images.githubusercontent.com/13002165/194623502-7ae1df91-7971-4d1b-a89f-18e140dfb040.jpg)

**Comment by lukasalexanderweber** [comment]
What I could Imagine is that if you have a really big amount of Images it would be good to stitch some parts of the scene seperately and then stitching those results together into a final Panorama.

**Comment by TByte007** [comment]
> What I could Imagine is that if you have a really big amount of Images it would be good to stitch some parts of the scene seperately and then stitching those results together into a final Panorama.

Is there a way to hint the algorithm where to search for the matching features ? Like if you have a map of like 10,000 x 10,000 images to tell it to search from 9000x0 to 10,000x768 and to match it with the new image or it's lot a more complicated than that ?

**Comment by lukasalexanderweber** [comment]
no I think you will be fine like having a grid of images and first stitch the "upper left" images, then "upper right"... and after that you can stitch the 4 results together, you just need to have enough overlap.

x  x 
x  x
